File: tests/visualtests/test_streamlines.py
import numpy as np
import matplotlib.pyplot as plt
from readmesh import read_off

radius = 0.2
#radius = 0.6
theta = np.linspace(0, 2*np.pi, 32)
cx = radius * np.cos(theta)
cy = radius * np.sin(theta)

ver, tri = read_off('../data/test_mesh2d.off')
#ver, tri = read_off('../data/test_annulus.off')
plt.triplot(ver[:, 0], ver[:, 1], tri, c=(0.8,)*3)
#plt.triplot(ver[:, 0], ver[:, 1], tri[[50, 100, 130]], c='r')

i = 0
with open('out', 'rt') as file:
    for line in file.readlines():
        x = np.fromstring(line, sep=' ').reshape((-1, 3))
        i += 1
        plt.plot(x[:, 0], x[:, 1], '.-')
        plt.plot(x[0, 0]+cx, x[0, 1]+cy, c=(0.7,)*3)
        plt.plot(x[-1, 0]+cx, x[-1, 1]+cy,  c=(0.7,)*3)
plt.axis('equal')
plt.show()

File: tests/visualtests/test_streamlines.cpp
#include "engine.cpp"
#include "readmesh.cpp"

int main()
{
    int nv, nt;
    double *ver, *orient;
    int *tri;
    char mesh_file[] = "../data/test_mesh2d.off";
    char orient_file[] = "../data/test_mesh2d.orient2";
    //char mesh_file[] = "../data/test_annulus.off";
    //char orient_file[] = "../data/test_annulus.orient";

    read_off(mesh_file, nv, nt, ver, tri);
    read_orient_txt(orient_file, nt, orient);

    StreamlineEngine engine;
    engine.initialize(nv, nt, ver, tri, orient, 0, 0);
    double radius = 0.1;
    //double radius = 0.6;

    /* setup arguments:
    double radius, int max_length, int max_nb_seeds, 
    int avoid_u_turns, double max_angle, 
    double singularity_mask_radius,
    unsigned int random_seed, int parallel, int num_threads */
    engine.setup(radius, 1024, 32, 1, 90, 0, 0.1, 129873, 0, -1);
    //int rgn[] = {50, 100, 130};
    //engine.define_seed_region(3, rgn);
    engine.run();
    
    for (int i=0;i<engine.collection.size();i++) {
        engine.collection[i]->print();
        printf("\n");
    }
}